include/igl/invert_diag.h

```diff
@@ -22,10 +22,10 @@ namespace igl
   //   X  an m by n sparse matrix
   // Outputs:
   //   Y  an m by n sparse matrix
-  template <typename T>
+  template <typename DerivedX, typename MatY>
   IGL_INLINE void invert_diag(
-    const Eigen::SparseMatrix<T>& X, 
-    Eigen::SparseMatrix<T>& Y);
```

Comment:
Same as above. It's bad practice to simply make things generic templates.

Reply:
> no member named 'coeffRef' in 'Eigen::SparseCompressedBase<Eigen::SparseMatrix<double, 0, int> >'